crates/aptos-crypto/src/bls12381/bls12381_validatable.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Aptos
```

Comment:
That's exactly what we are doing / will be doing.

First, the public key validation is only needed for normal, non-aggregated, non-multisig BLS signatures.

Second, verification of such signatures is currently only enabled in Move smart contracts (via this PR). Here, a contract would maintain some state about which PKs have been validated via `bls12381_validate_pubkey`. The cost for this call will be charged to the PK owner and not the signature verifier: for example, to register a PK, one would have to send a TXN that calls and pays for `bls12381_validate_pubkey`.

Once the contract validates a PK, it can rely on that PK to verify as many (normal, non-aggregated, non-multisig) signatures as it wants (without paying for any extra PK validations, which would be redundant and useless).

Third, once we add BLS as a transaction authenticators, we will do so via the `Validatable<bls12381::PublicKey>` struct. This implies the PK was validated before the account was registered. Again, the one-time validation cost will be paid by the user doing the registration and will never be incurred again during signature verification.